modal/functions.py

```diff
@@ -725,22 +727,25 @@ def call(self, *args, **kwargs) -> Awaitable[Any]:  # TODO: Generics/TypeVars
         else:
             return self._call_function(args, kwargs)
 
-    def __call__(self, *args, **kwargs) -> Any:  # TODO: Generics/TypeVars
-        if self._is_remote_cls_method:
-            return self.call(*args, **kwargs)
+    @synchronizer.nowrap
+    def __call__(wrapped_self, *args, **kwargs) -> Any:  # TODO: Generics/TypeVars
+        unwrapped_self = synchronizer._translate_in(wrapped_self)
+        if unwrapped_self._is_remote_cls_method:
+            # note this uses the wrapped self instead, since this is a remote version of the function
+            return wrapped_self.call(*args, **kwargs)
```

Comment:
It feels a bit weird that the behavior here is very different based on `_is_remote_cls_method` – this to me implies that either
1. Remote class methods are slightly different objects
2. Maybe we should get rid of `__call__` for non remote class methods like we talked about, and replace it with something like `.local(...)`? Since users are confused by this anyway.

Reply:
Yeah I agree. I still think the best would be to abolish the `remote()` instantiator and have everything use the same constructor, and then methods would work the same as function handles for local stub.functions, e.g.:
```
Foo(my_param=10).meth.call("hello")
```

But I think that would belong in a different PR, as this change is mostly just to make the new aio interface work as expected (this method is basically unchanged if you discount the variable renames and the nowrap-decorator)